### PR TITLE
Add scripts for working with `auto` repositories

### PR DESCRIPTION
**Pull request by jwodder**

### CODE REVIEW

File: auto/find-auto-dependa.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# /// script
```
Comment (yarikoptic) [inline auto/find-auto-dependa.py:2]: what handles such things?  might be worth adding a comment above. 

Ideally should be some way `pip` could understand directly somehow... I bet someone thought of extending pip with support for parsing some header out, did they?

Comment (yarikoptic) [reply auto/find-auto-dependa.py:2]: What about prepending all scripts which oriented toward operation with github with `gh-` prefix?

Comment (jwodder) [reply auto/find-auto-dependa.py:2]: This is [inline script metadata](https://packaging.python.org/en/latest/specifications/inline-script-metadata/) from PEP 723.  The only programs I know of that support it are pipx, pip-run, and hatch.

Comment (jwodder) [reply auto/find-auto-dependa.py:2]: > What about prepending all scripts which oriented toward operation with github with gh- prefix?

I don't see a need for that.

**Comment by yarikoptic** [comment]
coolio, let's merge them.

**Pull request merged**